codex/blockexchange/engine/engine.nim

```diff
@@ -581,7 +582,9 @@ proc blockexcTaskRunner(b: BlockExcEngine) {.async.} =
       peerCtx = await b.taskQueue.pop()
 
     trace "Got new task from queue", peerId = peerCtx.id
+    await peerCtx.lock.acquire()
```

Comment:
Can you explain why this is needed at all?

Reply:
Linked issue has some context https://github.com/codex-storage/nim-codex/issues/646